### ISSUE TITLE
Couldnt set it up on Ubuntu Server 18.04 

### ISSUE THREAD

**Issue by prithvihv** [fastai/fastai#799]
trying to run on cpu

```conda env update -f environment-cpu.yml```
 
pip : 18.0
python : 3.6
conda : 4.5.11

Stack Trace
```
Exception:
Traceback (most recent call last):
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2869, in _dep_map
    return self.__dep_map
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2663, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/packaging/requirements.py", line 93, in __init__
    req = REQUIREMENT.parseString(requirement_string)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 1632, in parseString
    raise exc
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 1622, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 1379, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 3395, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 1383, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pyparsing.py", line 3183, in parseImpl
    raise ParseException(instring, loc, self.errmsg, self)
pip._vendor.pyparsing.ParseException: Expected stringEnd (at char 33), (line:1, col:34)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2949, in __init__
    super(Requirement, self).__init__(requirement_string)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/packaging/requirements.py", line 97, in __init__
    requirement_string[e.loc:e.loc + 8]))
pip._vendor.packaging.requirements.InvalidRequirement: Invalid requirement, parse error at "'; extra '"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_internal/basecommand.py", line 141, in main
    status = self.run(options, args)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 299, in run
    resolver.resolve(requirement_set)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_internal/resolve.py", line 102, in resolve
    self._resolve_one(requirement_set, req)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_internal/resolve.py", line 306, in _resolve_one
    set(req_to_install.extras) - set(dist.extras)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2826, in extras
    return [dep for dep in self._dep_map if dep]
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2871, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2881, in _compute_dependencies
    reqs.extend(parse_requirements(req))
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2942, in parse_requirements
    yield Requirement(line)
  File "/home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2951, in __init__
    raise RequirementParseError(str(e))
pip._vendor.pkg_resources.RequirementParseError: Invalid requirement, parse error at "'; extra '"

 ```

**Comment by phromo** [comment]
I have the same issue using fully upgraded ubuntu 16.04 running conda 4.5.11 with python 3.6.6

It seems the pip installed into the fastai environment somehow is broken (any pip command or install attempt of any package fails with the same error as the original report). Setuptools is version 40.4.0 (conda-forge), pip is 18.0 (conda-forge)

**Comment by shan224** [comment]
Also have the same issue

**Comment by shan224** [comment]
Thanks @phromo but removing `conda-forge::jupyter_contrib_nbextensions` doesn't fix the issue for me. Has this fixed the issue for anyone else? 

The issue seems to be associated with the pip dependencies specifically in the environment.yml ( I deleted everything except for a single pip dep and it failed)

**Comment by phromo** [comment]
@shan224 did you remove your broken environment first? trying update on an already broken env won't work.
I just verified that my workaround does what's intended on a fresh ubuntu 1804, also verified that an unmodified fastai repo breaks with the error message above.

**Comment by prithvihv** [comment]
still not completely fixed, could somebody help me out ?
```Requirement already satisfied: click<8,>=4.0 in /home/prithvihv/anaconda3/envs/fastai-cpu/lib/python3.6/site-packages (from cligj>=0.4->fiona->geopandas>=0.3.0->plotnine->-r /home/prithvihv/AI/fastai/condaenv.6f2se111.requirements.txt (line 10)) (6.7)
twisted 18.7.0 requires PyHamcrest>=1.9.0, which is not installed.
mkl-random 1.0.1 requires cython, which is not installed.
spacy 2.0.12 has requirement regex==2017.4.5, but you'll have regex 2018.7.11 which is incompatible.
kaggle-cli 0.12.13 has requirement lxml<4.1,>=4.0.0, but you'll have lxml 4.2.5 which is incompatible.
```

**Comment by Hyaxia** [comment]
Any idea how to solve the issue if it happends with the appveyor tests?
https://github.com/jupyter/notebook/pull/4036
Thanks!

**Comment by nikogamulin** [comment]
Here I am pasting the [answer from the forum](http://forums.fast.ai/t/conda-env-update-error-on-ubuntu-16-04/23570/7?u=brismith), related to Ubuntu 16.04 installation. I believe the issue on 18.04 could be solved the same way

At first I just removed fastai repository, cloned it and the same error reappeared.

Then I ran conda env remove -y -n fastai before conda env update and there were no errors (haven’t modified environment.yml file as there have been already changes pushed to fastai repository).

Now the only warnings are the following:

> twisted 18.7.0 requires PyHamcrest>=1.9.0, which is not installed.
> mkl-random 1.0.1 requires cython, which is not installed.
> spacy 2.0.12 has requirement regex==2017.4.5, but you’ll have regex 2018.7.11 which is incompatible.

Afterward,s I have installed mkl-random and twisted with pip:
```
pip install twisted
pip install mkl-random
```

**Comment by shan224** [comment]
@phromo thanks. I didn't think the env was successfully created...after I removed it the env builds fine. 

@prithvihv, add twisted and mkl-random to your environment.yml dependencies and rerun `conda env update`

**Comment by phromo** [comment]
Changes happening in the fastai repo also to deal with the broken jupyter packages, see for instance commit 7528b2af1b7a0224225082e69f2b019209ffc1cd 

My workaround is now obsolete ;)

**Comment by Hyaxia** [comment]
In my case, I have found that the nbconvert is the problem.
When installing it with `conda install` it breaks pip o_O
So i just install it will `pip install` instead.

For more details see https://github.com/jupyter/notebook/pull/4036/commits/4c1d62f3b21af6f139d8396081578d0c1646a632

Cheers

**Comment by doranbae** [comment]
Hi, I still have a problem. Here's a list of things I tried.
0. set up config: ubuntu 16.04 running conda 4.5.11 with python 3.7.0
1. Using this version: https://github.com/udacity/CarND-Term1-Starter-Kit/blob/master/environment-gpu.yml
2. The above doesn't have `conda-forge::jupyter_contrib_nbextensions` , so I didn't try removing this line from the yml file.
3. I tried `conda env remove -n carnd-term1` and then again `conda env create -f environment-gpu.yml`. 

Even after this, I am still getting the same error message:

```
Exception:
Traceback (most recent call last):
  File "/home/dorandoran_hk/miniconda3/envs/carnd-term1/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2869, in _dep_map
    return self.__dep_map
  File "/home/dorandoran_hk/miniconda3/envs/carnd-term1/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2663, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:
```
Any help will be appreciated!

**Comment by doranbae** [comment]
Hi, I will delete this if my post is inappropriate. I am also running into the same problem however, it is not related to fastai directly. I apologize if this is inappropriate place to post this question, if you let me know, i will delete it. but i was out of options so I decided to ask here. I am trying to set up conda environment and running `conda env update -f environment-cpu.yml` gives me the following error. 


0. set up config: ubuntu 16.04 running conda 4.5.11 with python 3.7.0
1. Using this version: https://github.com/udacity/CarND-Term1-Starter-Kit/blob/master/environment-gpu.yml
2. The above doesn't have `conda-forge::jupyter_contrib_nbextensions` , so I didn't try removing this line from the yml file.
3. I tried `conda env remove -n carnd-term1` and then again `conda env create -f environment-gpu.yml`. 

Even after this, I am still getting the same error message:

```
Exception:
Traceback (most recent call last):
  File "/home/dorandoran_hk/miniconda3/envs/carnd-term1/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2869, in _dep_map
    return self.__dep_map
  File "/home/dorandoran_hk/miniconda3/envs/carnd-term1/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2663, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:
```
Any help will be appreciated!
